cmd/agent/config/config.go

```diff
@@ -0,0 +1,95 @@
+package config
+
+import (
+	"encoding/json"
+	"flag"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/caarlos0/env/v6"
+)
+
+type envConfig struct {
+	Config string `env:"CONFIG"`
+}
+
+// AgentConfig структура для парсинга файла конфигурации
+type AgentConfig struct {
+	Address        string `json:"address"`
+	ReportInterval string `json:"report_interval"`
+	PollInterval   string `json:"poll_interval"`
+	HashKey        string `json:"hash_key"`
+	RateLimit      string `json:"rate_limit"`
+	CryptoKey      string `json:"crypto_key"`
+}
+
+// loadConfig загружает конфигурацию из файла
+func (cfg *AgentConfig) loadConfig(path string) error {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return err
```

Comment:
Ошибки лучше оборачивать в дополнительный контекст:

```suggestion
		return fmt.Errorf("failed to read the config file %s: %w", path, err)
```

Reply:
вроде поправил